appinventor/components/src/com/google/appinventor/components/runtime/Clock.java

```diff
@@ -207,6 +216,96 @@ public static Calendar MakeInstant(String from) {
     }
   }
 
+  /**
+   * An instant in time specified by number year, number month, number day
+   * @param year year integer
+   * @param month month integer
+   * @param day day integer
+   * @return  Calendar instant
+   */
+  @SimpleFunction(description = "Allows the user to set the clock to be "
+    +"a date value.\n" +
+    "Valid values for the month field are 1-12 and 1-31 for the day field.\n")
+  public Calendar MakeDate(int year, int month, int day) {
+    int jMonth = month - 1;
+    try {
+      GregorianCalendar cal = new GregorianCalendar(year, jMonth, day);
+      cal.setLenient(false);
+
+      // A non-lenient GregorianCalendar throws an exception upon 
+      // calculating its time or calendar field values if any out-of-range field value has been set.
+      cal.getTime();
+    } catch (IllegalArgumentException e) {
+      form.dispatchErrorOccurredEvent(this, "MakeDate", ErrorMessages.ERROR_ILLEGAL_DATE);
+    }
+    
+    Calendar instant = Dates.DateInstant(year, month, day);
+    return instant;
+  }
+
+  /**
+   * An instant in time specified by integer hour and integer minute
+   * @param hour hour integer
+   * @param minute minute integer
+   * @return  Calendar instant since 1/1/1970
+   */
+  @SimpleFunction(description = "Allows the user to set the time of the clock - " +
+    "Valid format is hh:mm\n")
+  public Calendar MakeTime(int hour, int minute) {
```

Comment:
@xueyuanw I went back to test this one last time and realized that we might want to allow for seconds here. You should update the description and documentation accordingly.